File: backend/gemini_wrapper.py
import os
from google import genai
from dotenv import load_dotenv

load_dotenv()

api_key = os.getenv("GEMINI_API_KEY")

# Configure the client using an environment variable
client = genai.Client(api_key=api_key)

def generate_explanation(prompt: str, api_key: str) -> str | None:
    try: 
        # Use the provided API key instead of environment variable
        client = genai.Client(api_key=api_key)
        response = client.models.generate_content(model="gemini-1.5-flash", contents=prompt)
        
        if response is None:
            return "Unable to generate explanation"
        
        return response.text
    except Exception as e:
        print(f"Error generating explanation: {e}")
        return f"Error generating explanation: {str(e)}"

File: gemini_wrapper.js
const { GoogleGenerativeAI } = require("@google/generative-ai");

async function generate_explanation(prompt, apiKey) {
  try {
    if (!apiKey) {
      throw new Error("API key is required");
    }

    const genAI = new GoogleGenerativeAI(apiKey);
    const model = genAI.getGenerativeModel({ model: "gemini-2.5-flash-lite" });

    const result = await model.generateContent(prompt);
    const response = await result.response;
    const text = response.text();

    if (!text) {
      return "Unable to generate explanation";
    }

    return text;
  } catch (error) {
    console.error(`Error generating explanation: ${error}`);
    return `Error generating explanation: ${error.message}`;
  }
}

module.exports = { generate_explanation };
